config/defaults.go

```diff
@@ -80,6 +87,10 @@ func (d *Defaults) Validate() error {
 		}
 	}
 
+	if d.KillAfter == -1 {
+		return errors.Errorf("killAfter missing")
+	}
+
```

Comment:
For flag passing a duration, I don't have the chance to leave it empty, I have to give it some default value, I set up the default value to be negative to make sure we catch the error. Is this a too hacky?

Reply:
Can't it be zero, or something like that? The question would be if we want to always require a 'killafter', or not. The behavior when killafter of 0 is passed to timeout is to not implement killafter. So I think 0 would have the correct behavior.

Reply:
But I see your point, it should be set to something and we probably don't want to let the user avoid setting it. So setting it to -1 sounds ok to me.

Reply:
> That is a good reason to make KillAfter be of type float64 (since in the end we are going to use it as a float), and just put in the help string that it is "in seconds".

For the default value, you could use `math.NaN()`, and then check if it was set with `math.IsNaN(d.KillAfter)`.

Reply:
Actually we should check the flag itself, data.KillAfter, with math.IsNan(), and perhaps also that it is not negative.  For d.KillAfter, we should just check that it is not negative perhaps.

Reply:
I think at the time we pass a values through flag, in configure.go file, missing input was checked with `flag.NArg() != 2`, it only allows 2 flag with no argument input, might be sufficient for missing input check? I added the check for d.KillAfter to be non-negative

Reply:
I tested the duration and it turns out "10" gives a parsing error, so I think we are safe in using Duration for the flag. The default value should probably be something like -time.Second (or -1 * time.Second if we want to be explicit).

Reply:
Let's go with what is there now, because it works and the type of the flag in configure.go is a minor implementation detail, not an interface.

Reply:
I tried the duration, "10s" gives a parsing error too, it is actually more convenient to use float64 than using Duration. I will just kept it with float64. Thank you for testing this with me.